com.unity.netcode.gameobjects/Tests/Runtime/DeferredMessagingTests.cs

```diff
@@ -0,0 +1,1230 @@
+        {
+        }
+
+        public void OnAfterSendBatch(ulong clientId, int messageCount, int batchSizeInBytes, NetworkDelivery delivery)
+        {
+        }
+
+        public void OnBeforeReceiveBatch(ulong senderId, int messageCount, int batchSizeInBytes)
+        {
+        }
+
+        public void OnAfterReceiveBatch(ulong senderId, int messageCount, int batchSizeInBytes)
+        {
+        }
+
+        public bool OnVerifyCanSend(ulong destinationId, Type messageType, NetworkDelivery delivery)
+        {
+            return true;
+        }
+
+        public bool OnVerifyCanReceive(ulong senderId, Type messageType, FastBufferReader messageContent, ref NetworkContext context)
+        {
+            if (messageType == typeof(CreateObjectMessage))
+            {
+                CaughtMessages.Add(new TriggerData
+                {
+                    Reader = new FastBufferReader(messageContent, Allocator.Persistent),
+                    Header = context.Header,
+                    Timestamp = context.Timestamp,
+                    SenderId = context.SenderId,
+                    SerializedHeaderSize = context.SerializedHeaderSize
+                });
+                return false;
+            }
+
+            return true;
+        }
+
+        public void OnBeforeHandleMessage<T>(ref T message, ref NetworkContext context) where T : INetworkMessage
+        {
+        }
+
+        public void OnAfterHandleMessage<T>(ref T message, ref NetworkContext context) where T : INetworkMessage
+        {
+        }
+    }
+
+    public class DeferredMessageTestRpcComponent : NetworkBehaviour
+    {
+        public bool ClientRpcCalled;
+
+        [ClientRpc]
+        public void SendTestClientRpc()
+        {
+            ClientRpcCalled = true;
+        }
+    }
+
+    public class DeferredMessageTestNetworkVariableComponent : NetworkBehaviour
+    {
+        public NetworkVariable<int> TestNetworkVariable = new NetworkVariable<int>();
+    }
+
+    public class DeferredMessageTestRpcAndNetworkVariableComponent : NetworkBehaviour
+    {
+        public bool ClientRpcCalled;
+
+        [ClientRpc]
+        public void SendTestClientRpc()
+        {
+            ClientRpcCalled = true;
+        }
+
+        public NetworkVariable<int> TestNetworkVariable = new NetworkVariable<int>();
+    }
+
+    public class DeferredMessagingTest : NetcodeIntegrationTest
+    {
+        protected override int NumberOfClients => 2;
+
+        private List<SpawnCatcher> m_ClientSpawnCatchers = new List<SpawnCatcher>();
+
+        private GameObject m_RpcPrefab;
+        private GameObject m_NetworkVariablePrefab;
+        private GameObject m_RpcAndNetworkVariablePrefab;
+
+        protected override IEnumerator OnSetup()
+        {
+            // Host is irrelevant, messages don't get sent to the host "client"
+            m_UseHost = false;
+
+            m_RpcPrefab = new GameObject("Object");
```

Comment:
name the object better? :P

Reply:
Heh, this is literally what every object we make as a prefab is named. XD Does the name matter? It's not referenced anywhere.

Reply:
might help when debugging issues during testing but definitely just nit-picks/minors here.